python/ycm/vimsupport.py

```diff
@@ -575,3 +575,68 @@ def SearchInCurrentBuffer( pattern ):
 
 def LineTextInCurrentBuffer( line ):
   return vim.current.buffer[ line ]
+
+
+def ClosePreviewWindow():
+  """ Close the preview window if it is present, otherwise do nothing """
+  vim.command( 'silent! pclose!' )
+
+
+def JumpToPreviewWindow():
+  """ Jump the vim cursor to the preview window, which must be active. Returns
+  boolean indicating if the cursor ended up in the preview window """
+  vim.command( 'silent! wincmd P' )
+  return vim.current.window.options[ 'previewwindow' ]
+
+
+def JumpToPreviousWindow():
+  """ Jump the vim cursor to its previous window position """
+  vim.command( 'silent! wincmd p' )
+
+
+def OpenFileInPreviewWindow( filename ):
+  """ Open the supplied filename in the preview window """
+  vim.command( 'silent! pedit! ' + filename )
+
+
+def WriteToPreviewWindow( message ):
+  """ Display the supplied message in the preview window """
+
+  # This isn't something that comes naturally to Vim. Vim only wants to show
+  # tags and/or actual files in the preview window, so we have to hack it a
+  # little bit. We generate a temporary file name and "open" that, then write
+  # the data to it. We make sure the buffer can't be edited or saved. Other
+  # approaches include simply opening a split, but we want to take advantage of
+  # the existing Vim options for preview window height, etc.
+
+  ClosePreviewWindow()
+
+  # Generate a temporary file name - we don't actually create the file
```

Comment:
This seems covered by the big comment block above, so in a sense this just duplicates that unnecessarily.